### ISSUE TITLE
luacheck: Fails due to cache write race condition

### ISSUE THREAD

**Issue by hunger** [lunarmodules/luacheck#52]
I am trying to use luacheck in pre-commit.ci and get a lot of false positives there.

This is the output I get:

```
[21 lines ending in ok]
Total: 0 warnings / 0 errors in 21 files
Critical error: Couldn't initialize cache in /tmp/cache/luacheck: Couldn't make directory /tmp/cache/luacheck: File exists
```

The devs over at pre-commit looked into the issue and came to the conclusion that luacheck is racing with its cache directory creation.

> I can reproduce the same thing locally by repeatedly trying this command:
>
> rm -rf ~/.cache/luacheck; PATH=$PWD/opt/lua/bin/:$PATH pre-commit run luacheck --all-files
> please report the issue to luacheck

pointing to this code: https://github.com/lunarmodules/luacheck/blob/81e0835cf5b54b9cb2101b21edeec011dec5fff3/src/luacheck/fs.lua#L177-L201

This is the issue I raised with pre-commit.ci: https://github.com/pre-commit-ci/issues/issues/114

**Comment by alerque** [comment]
Yes I see what the issues is ... Unfortunately the bug is even upstream of us, technically the [lfs project](https://github.com/keplerproject/luafilesystem) needs a fix, anything else we can do is just a hack around the missing option there (raised [upstream issue here](https://github.com/keplerproject/luafilesystem/issues/156)).

In the mean time we can probably fudge and catch this particular error state, check again to see why we we ended up there, then hope for the best and go on. This isn't ideal though and I would suggest part of the problem is `luacheck` should be allowed to manage it's own threads instead of having them imposed. I'll raise that on the pre-check issue.
